### DISCUSSION TITLE
how to integrate sveltia cms into sveltekit website?

### DISCUSSION THREAD

**Discussion by NVCDevelopmentTeam** [sveltia/sveltia-cms#256] [Q&A]
Hello, I'm building a website using sveltekit but I can't integrate sveltia cms into my website. Although I have configured it to suit my website, when I log in to sveltia cms, it gets a 404 error and I can't log in. Perhaps, due to oauth or some problem in sveltia cms. I hope you can help me on this issue. Thanks, here is my repo: [Click here](https://github.com/NVCDevelopmentTeam/Zepp-OS-ScreenReader.git) Website location is in directory /site/htdocs.

**Comment by kyoshino** [comment]
Looks like you haven’t installed an OAuth client. Here you have [`auth_endpoint: oauth`](https://github.com/NVCDevelopmentTeam/Zepp-OS-ScreenReader/blob/a6aaa74c8a317633af230c5538bcebde162d5eb2/site/htdocs/static/admin/config.yml#L7C18-L7C23) but you don’t have the `oauth` route. You can use [our own OAuth client](https://github.com/sveltia/sveltia-cms-auth) or [any other 3rd party client](https://decapcms.org/docs/external-oauth-clients/) made for Netlify/Decap CMS.

`sveltia-cms-auth` is built for Cloudflare Workers, but maybe you can make it work with Vercel Functions by modifying the [source](https://github.com/sveltia/sveltia-cms-auth/blob/main/src/index.js) a bit. (I can’t try as I don’t have a Vercel account.)

**Reply by kyoshino** [reply]
Alternatively, you can [work with a local repository](https://github.com/sveltia/sveltia-cms?tab=readme-ov-file#working-with-a-local-git-repository) and push changes manually to GitHub. In that case, you don’t need to install an OAuth client. Just run `npm run dev` on your SvelteKit project, open `http://localhost:5173/admin/index.html` with Chrome, and you’ll be able to edit your content without signing in with GitHub.

**Comment by NVCDevelopmentTeam** [comment]
==Does it support sveltekit? As for the configuration, I think it's quite good, but it sounds like it's only suitable for netlify. You can check out my repo to help me configure everything to match the project. Then I'll fix it later.

**Comment by NVCDevelopmentTeam** [comment]
Instead, I want to directly access my website link instead of having to use localhost for administrative convenience, because I am blind so I need stability.

**Comment by kyoshino** [comment]
These OAuth clients are indented for use outside of Netlify. [`sveltia-cms-auth`](https://github.com/sveltia/sveltia-cms-auth) is a framework-agnostic Cloudflare Workers script. As I said, it can probably be modified to work with Vercel Functions, perhaps as part of a SvelteKit project, but I’m not sure. Unfortunately, I cannot help you set it up.

**Comment by NVCDevelopmentTeam** [comment]
Can it be installed using npm command? I wonder if the number of languages ​​of sveltia cms depends on the website because I see that its language list only has two languages? If it is used for Cloudflare, is it possible that vercel will not be able to run?

**Reply by kyoshino** [reply]
No, an OAuth client cannot be installed with npm. It’s basically a Node.js script independent from the CMS. I don’t have a Vercel account so I cannot tell you how difficult it is to modify and deploy it to Vercel.

**Comment by NVCDevelopmentTeam** [comment]
If necessary, you can give specific instructions so that I can apply it to the java script to complete it.

**Comment by NVCDevelopmentTeam** [comment]
If it is not installed via npm then how is it installed?

**Reply by kyoshino** [reply]
The [README](https://github.com/sveltia/sveltia-cms-auth/blob/main/README.md) has all the information.

**Comment by NVCDevelopmentTeam** [comment]
It only supports Cloudflare and netlify but vercel may not be in this list. That means I have to clone it before installing it, right?

**Reply by kyoshino** [reply]
Sveltia CMS Authenticator only works with CloudFlare Workers.

If you use Netlify, you don’t have to install it as Netlify facilitates the authentication flow.

If you use Vercel, yes, you need to clone it first and probably need to modify it to make it work with Vercel. As I already said, I cannot help that part.

**Comment by NVCDevelopmentTeam** [comment]
I tried setting it up but when I logged in it showed api.netlify.com and not github so I couldn't log in to it. IS it that I'm missing something or misconfiguring oauth?

**Comment by NVCDevelopmentTeam** [comment]
The problem is, after cloning your project, which folder of the project will it be placed in?

**Comment by NVCDevelopmentTeam** [comment]
I placed it in the website folder but it's not compatible.

**Comment by kyoshino** [comment]
I cannot help you set up the stuff.

**Comment by NVCDevelopmentTeam** [comment]
You just need to edit the code properly and I will do the deploying configuration. Otherwise, you must create a configuration for vercel and specify the file that needs to be edited. One thing I wonder is that if my personal website does not have the user authorization feature, they will illegally log in to my website. This,affects security. If you use it on localhost, you have to do one more step and commit the code, which takes a lot of time. Not to mention, I participate in adsense so it will affect the website.
